File: iosApp/iosApp/Models/Recipe/Recipe.swift
//
//  Recipe.swift
//  BauchGlück
//
//  Created by Frederik Kohler on 23.07.24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import SwiftUI

struct Recipe: Identifiable, Codable {
    var id: String
    var user_id: String
    var title: String
    var recipeCategory: RecipeCategories
    var portion_size: String
    var preparation_time: String
    var cooking_time: String
    var ingredients: [Ingredients]
    var preparation: String
    var rating: Int
    var notes: String
    var image: String
    var is_private: Bool
    var last_updated: Date
}

File: shared/src/commonMain/kotlin/model/recipe/Recipe.kt
package model.recipe

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class Recipe(
    var id: String,
    var userID: String,
    var title: String,
    var recipeCategory: RecipeCategory,
    var portionSize: String,
    var preparationTime: String,
    var cookingTime: String,
    var ingredients: List<Ingredient>,
    var preparation: String,
    var rating: Int,
    var notes: String,
    var image: String,
    var isPrivate: Boolean,
    var created: String,
    var lastUpdated: String,
)